src/compiler/filesystem_local_win.cc

```diff
@@ -44,45 +44,63 @@ char* FilesystemLocal::get_executable_path() {
   return path;
 }
 
-bool FilesystemLocal::is_absolute(const char* path) {
-  if (SourceManager::is_virtual_file(path)) return true;
+// Returns the size of the root prefix of the given path.
+// Returns 0 if the path doesn't have any root prefix.
+// Accepted roots are:
+// - drives: `c:\` and `c:/`. We do not accept 'c:' here.
+// - posix roots `/`
+// - the double '\\' of a network path: `\\Machine1` or `\\wsl$`. In this
+//   case we consider `\\` to be the root path.
+//   Contrary to the file path (like "c:\") the returned root includes more
+//   than just one drive, but that's more in spirit with the original root
+//   path anyway.
+// - a virtual file, with the VIRTUAL_FILE_PREFIX.
+static int root_prefix_length(const char* path) {
+  if (SourceManager::is_virtual_file(path)) {
+    return strlen(SourceManager::VIRTUAL_FILE_PREFIX);
+  }
   int length = strlen(path);
-  if (length < 3) return false;
-  // Either a Windows drive like "c:\", or a network path "\\Machine1".
-  // Network paths also include the WSL drive.
-  return (path[1] == ':' && is_path_separator(path[2])) ||
-      (is_path_separator(path[0]) && is_path_separator(path[1]));
+  if (length == 0) return 0;
+  if (path[0] == '/') {
+    if (length == 1) return 1;  // Posix root.
+    if (!is_path_separator(path[1])) return 1;  // Posix root prefix.
```

Comment:
So /\ is a network path prefix?

Reply:
Had to change this.
Posix-style roots aren't really absolute, as they are relative to the current drive.